roles/zuul/tasks/main.yml

```diff
@@ -136,6 +136,17 @@
   include: merger.yml
   when: "'zuul-merger' in zuul_components"
 
+- name: Stop unwanted zuul services
+  service:
+    name: "{{ item }}"
+    state: stopped
+    enabled: no
+  when: "item not in zuul_components"
+  with_items:
+    - zuul-launcher
+    - zuul-merger
+    - zuul-server
```

Comment:
My concern with this is that this module will throw an error if the service doesn't exist on the box. At Blue Box we had a version of the module that took an extra option that ignored the service being missing. I'd rather not carry that over here, but instead construct a failed_when setup that handles the scenario.